lib/main.dart

```diff
@@ -12,12 +12,17 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import 'package:firebase_core/firebase_core.dart';
 import 'package:flutter/material.dart';
 
 import 'src/app.dart' deferred as app;
 
-void main() {
-  final Future<void> loadedLibrary = app.loadLibrary();
+void main() async {
+  // Initialize Firebase
+  WidgetsFlutterBinding.ensureInitialized();
```

Comment:
Is this required? I've only seen this in tests?

Reply:
That implementation comes from [this SO answer](https://stackoverflow.com/a/63492262) (before I used [this from the offic docs](https://firebase.flutter.dev/docs/overview#initializing-flutterfire)). I think it ensures that the Flutter engine is activated so platform-specific code can be called

It is also used as an example [in pub.dev](https://pub.dev/packages/cloud_firestore/example)

Reply:
Ah, ok, I understand now. I wasn't seeing that you're calling that in `main`, and not somewhere inside the App itself!